lib/processors/debugFileCreator.js

```diff
@@ -8,12 +9,41 @@ const copier = require("./resourceCopier");
  * @param {Resource[]} parameters.resources List of resources to be processed
  * @returns {Promise<Resource[]>} Promise resolving with debug resources
  */
-module.exports = function({resources}) {
-	return copier({
-		resources: resources,
-		options: {
-			pattern: /((\.view|\.fragment|\.controller)?\.js)/,
-			replacement: "-dbg$1"
-		}
+module.exports = function({resources, fs}) {
+	const options = {
+		pattern: /((\.view|\.fragment|\.controller)?\.js)/,
+		replacement: "-dbg$1"
+	};
+
+	const stat = util.promisify(fs.stat);
+
+	return Promise.all(
+		resources.map((resource) => {
+			// check whether the debug resource path is already used in the
+			// previous tasks
+			return stat(resource.getPath().replace(options.pattern, options.replacement))
+				.then(
+					// if the file can be found, it should be filtered out from createing debug file
+					() => false,
+					(err) => {
+						if (err.code === "ENOENT") {
+							// if the file can't be found, it should be included in creating debug file
+							return resource;
+						}
+						// if it's other error, forward it
+						throw err;
+					}
+				);
+		})
+	).then((stats) => {
```

Comment:
What about `results` instead of `stats`, as the array doesn't contain any stat info?

Reply:
done